include/zephyr/net/sntp.h

```diff
@@ -88,6 +89,52 @@ int sntp_query(struct sntp_ctx *ctx, uint32_t timeout,
  */
 void sntp_close(struct sntp_ctx *ctx);
 
+/**
+ * @brief Initialise SNTP context for async operation
+ *
+ * Asynchronous operation is powered by @kconfig{CONFIG_NET_SOCKETS_SERVICE}.
+ *
+ * @param service Socket service defined by @ref NET_SOCKET_SERVICE_SYNC_DEFINE
+ * @param ctx Address of sntp context.
+ * @param addr IP address of NTP/SNTP server.
+ * @param addr_len IP address length of NTP/SNTP server.
+ *
+ * @return 0 if ok, <0 if error.
+ */
+int sntp_init_async(const struct net_socket_service_desc *service,
```

Comment:
I believe the first argument should be the context which is being initialized, with the service being moved to be the last argument. This will help the user understand that this is an alernative to the "normal" init :)